MediaBrowser.Common/Json/Converters/JsonNullableStructConverter.cs

```diff
@@ -32,7 +33,18 @@ public JsonNullableStructConverter(JsonConverter<T?> baseJsonConverter)
                 return null;
             }
 
-            return _baseJsonConverter.Read(ref reader, typeToConvert, options);
+            try
+            {
+                return _baseJsonConverter.Read(ref reader, typeToConvert, options);
+            }
+            catch
+            {
+                // This is a quoted number. Get the value as a string, then convert it manually.
```

Comment:
Rather than catching all exceptions, you can check the current token type of the reader:

```cs
if (reader.TokenType == JsonTokenType.String)
{
    // Handle as string
}
else if (reader.TokenType == JsonTokenType.Number)
{
    // Handle as number
}
else
{
    // Serializer catches this and rethrows a nice exception including JSON path and line number position
    // throw new JsonException();
}
```

Reply:
Things can always be improved - and your code is better.

Unless someone says otherwise, I'm going to remove the generics from this function and create two separate convertors so that the code can be simplified.

Reply:
Turns out reader.Token can also be null and the value be valid, so JsonTokenType.Number is the default assumed type in the current code.